src/bin/leftwm.rs

```diff
@@ -51,3 +64,38 @@ fn main() {
         }
     }
 }
+
+/// Executes a subcommand.
+///
+/// If a valid subcommand is supplied, executes that subcommand, passing `args` to the program.
+/// Valid subcommands are `check`, `command`, `state` and `theme`.
+/// Prints an error to STDERR and exits non-zero if an invalid subcommand is supplied, or there is
+/// some error while executing the subprocess.
+///
+/// # Arguments
+///
+/// + `args` - The command line arguments leftwm was called with.
+///
+/// # Panics
+///
+/// Panics if `args` has length < 2.
+///
+/// # Exits
+///
+/// Exits 1 if the first argument is not a valid subcommand.
+/// Exits 2 if the first argument is a valid subcommand, but the associated program failed to run.
+fn execute_subcommand(args: Vec<String>) {
+    let subcommands = ["check", "command", "state", "theme"];
+    // If the second argument is a valid subcommand
+    if subcommands.iter().any(|x| x == &args[1]) {
```

Comment:
Might want to do lower case check in case someone types for example leftwm Command? Not sure what proper shell behaviour is.

Reply:
I think if someone types `leftwm Command` it will print `Invalid command 'Command'`. I think that is acceptable behavior.

Reply:
If I type `leftwm-Command` I get `zsh: command not found: leftwm-Command` which makes perfect sense.

Reply:
Fair enough. That makes sense, was more so interested in how others handle sub commands, but they do the same.